ktane/modules/buttons.py

```diff
@@ -0,0 +1,139 @@
+# class and into a more general space.
+# Quitting in the middle of an event is also not functional because you can't
+# correctly unravel the stack with that input unless you account for it
+# with every function call (which defeats the purpose).
+class Buttons(Module):
+
+    # Number of Batteries
+    batteries = -1
+
+    # Lit indicators
+    lit_indicators = ['None']
+
+    # Entry point for the module
+    def run(self):
+
+        # Reset Number of Batteries
+        self.batteries = -1
+
+        # Reset Lit indicators
+        self.lit_indicators = ['None']
+
+        # Get the button
+        color = Color[self.get_valid_input_or_quit(
+            lambda x: x.lower() in Color.__members__.keys(),
+            "Enter the color of the button: ")]
+
+        text = Text[self.get_valid_input_or_quit(
+            lambda x: x.lower() in Text.__members__.keys(),
+            "Enter the button text: ")]
+
+        self.output_and_wait(self.apply_logic(color, text))
+
+    # Function to apply the logic for solving the button based on inputs
+    def apply_logic(self, color, text):
+        # Case 1
+        if (color is Color.blue and text is Text.abort):
+            return self.hold()
+
+        # Case 2
+        if (text is Text.detonate and self.get_batteries() > 1):
+            return self.press_and_release()
+
+        # Case 3
+        if (color is Color.white and "CAR" in self.get_lit_indicators()):
+            return self.hold()
+
+        # Case 4
+        if (self.get_batteries() > 2 and "FRK" in self.get_lit_indicators()):
+            return self.press_and_release()
+
+        # Case 5
+        if (color is Color.yellow):
+            return self.hold()
+
+        # Case 6
+        if (color is Color.red and text is Text.hold):
+            return self.press_and_release()
+
+        # Case 7
+        return self.hold()
+
+    # Press and release
+    def press_and_release(self):
+        return "Press and immediately release the button"
+
+    def hold(self):
+        print ('\nHold the button...\n')
+        stripe = Color[self.get_valid_input_or_quit(
+            lambda x: x.lower() in Color.__members__.keys(),
+            "Enter the color of the the stripe: ")]
+
+        return "Release when the timer has a {} in any position".format(self.get_stripe_number(stripe))
+
+    # Function to get countdown timer number from stripe color
+    def get_stripe_number(self, stripe):
+        if (stripe is Color.blue):
+            return 4
+        elif (stripe is Color.yellow):
+            return 5
+        else:
+            return 1
+
+    # Function to get number of batteries from user
+    def get_batteries(self):
+        if (self.batteries is -1):
+            batts = self.get_number_or_quit(
+                lambda x: x >= 0,
+                "Enter number of batteries: "
+                )
+            self.batteries = batts
+            return self.batteries
+        else:
+            return self.batteries
```

Comment:
in all cases you return self.batteries. this means you can place this outside the 'if' and have no 'else.'

Also, you dont need that temporary 'batts' variable. just self.batteries = self.get_number_or_quit(...)